shelly/99-shelly.js

```diff
@@ -337,4 +337,103 @@ module.exports = function (RED) {
             password: { type: "password" },
         }
     });
+    // --------------------------------------------------------------------------------------------
+    // The dimmer node controls a shelly dimmer.
+    function ShellyDimmerNode(config) {
+        RED.nodes.createNode(this, config);
+        var node = this;
+        node.hostname = config.hostname;
+
+        /* node.shellyInfo
+        GET /shelly
+        {
+           "type":"SHDM-1",
+           "mac":"CC50E3F36XXX",
+           "auth":false,
+           "fw":"20200309-104554/v1.6.0@43056d58",
+           "num_outputs":1,
+           "num_meters":1
+        }
+        */
+       shellyGet('/shelly', node, function(result) {
+            node.shellyInfo = JSON.parse(result);
+            if(node.shellyInfo.type.startsWith("SHDM-")){
+                node.status({ fill: "green", shape: "ring", text: "Connected." });
+            }
+            else{
+                node.status({ fill: "red", shape: "ring", text: "Shelly type " + node.shellyInfo.type + " is not known." });
+            }
+        });
+
+        /* when a payload is received in the format
+            {
+                light : 0,
+                on : true,
+                brightness : 75
+            }
+        then the command is send to the shelly.
+
+        The output gets the status of all relays.
+        */
+        this.on('input', function (msg) {
+
+            var route;
+            if(msg.payload !== undefined){
+                var command = msg.payload;
+
+                var light = 0;
+                if(command.light !== undefined){
+                    light = command.light;
+                }
+
+                var onOff;
+                if(command.on !== undefined){
+                    if(command.on == true){
+                        onOff = "on";
+                    }
+                    else{
+                        onOff = "off"
+                    }
+                }
+
+                var brightness;
+                if(command.brightness !== undefined){
+                    if(command.brightness >=1 && command.brightness <= 100){
+                        brightness = command.brightness;
+                  }

```

Comment:
The else is missing: what should happen here? Is that worth an error or why do we ignore it??